rose/client/main.py

```diff
@@ -71,27 +70,23 @@ def send_message(self, msg):
         self.client.sendLine(str(msg).encode('utf-8'))
 
 
-def load_driver_module(file_path):
+def load_driver_module(driver_path):
     """
-    Load the driver module from file
+    Load the driver module from the specified path.
 
-    Arguments:
-      file_path (str): The path to the driver module
-
-    Returns:
-        Driver module (module)
-
-    Raises:
-        ImportError if the module cannot be loaded
+    :param driver_path: Path to the driver module.
+    :return: The loaded module.
+    :raises ImportError: If there's an issue loading the module.
     """
-    module_dir, module_name = os.path.split(os.path.splitext(file_path)[0])
-    fp, pathname, description = imp.find_module(module_name, [module_dir])
     try:
-        return imp.load_module(module_name, fp, pathname, description)
-    finally:
-        # Since we may exit via an exception, close fp explicitly.
-        if fp:
-            fp.close()
+        spec = importlib.util.spec_from_file_location(
+            "driver_module", driver_path)
+        driver_module = importlib.util.module_from_spec(spec)
+        spec.loader.exec_module(driver_module)
+        return driver_module
+    except Exception as e:
+        raise ImportError(
+            f"Error loading driver module from path {driver_path}: {str(e)}")
```

Comment:
Since we cannot handle this error, it is better not to handle it and let if propagate to the top level caller. Changing the error is also not related to removing python 2 leftovers so it is best to just replace usage of the imp modujle with the modren version and leave error handling as is. We can revisit error handling later if needed.

Reply:
yeap, makes sense to me too, remvoed